### PR TITLE
spanner: support SPANNER_EMULATOR_HOST

### PR DESCRIPTION
**Pull request by jadekler**

### CODE REVIEW

**Comment by jadekler** [comment]
@larkee

**Comment by larkee** [comment]
I looked into this some more and the admin clients do not support having both transport and credentials set. So you will also need to set credentials to None when the environment variable is set.

File: spanner/google/cloud/spanner_v1/client.py [hunk 1]
```diff
@@ -51,6 +61,11 @@
 )
 
 
+if os.getenv('SPANNER_EMULATOR_HOST') != None and \
+    ('http://' in os.getenv('SPANNER_EMULATOR_HOST') or 'https://' in os.getenv('SPANNER_EMULATOR_HOST')):
```
Comment (bvandiver) [inline spanner/google/cloud/spanner_v1/client.py]: Worth an abstraction like get_spanner_emulator_host() as opposed to bare getenv calls?

Comment (larkee) [reply spanner/google/cloud/spanner_v1/client.py]: Done.

**Comment by jadekler** [comment]
@larkee as I'm rolling off client libraries, this will need to be picked up by someone else. Could you help find the appropriate person to do so?

**Comment by larkee** [comment]
> @larkee as I'm rolling off client libraries, this will need to be picked up by someone else. Could you help find the appropriate person to do so?

I'll pick it up. Let me know if you'd rather I opened a new PR rather than working on this one.

**Comment by jadekler** [comment]
> > @larkee as I'm rolling off client libraries, this will need to be picked up by someone else. Could you help find the appropriate person to do so?
> 
> I'll pick it up. Let me know if you'd rather I opened a new PR rather than working on this one.

Whichever is your preference! :)

**Pull request closed without merging**